In [102]:
import pandas as pd
import numpy as np
# import matplotlib.pyplot as plt
# import seaborn as sns

from time import strptime, strftime
from datetime import datetime

In [103]:
df= pd.read_csv("./archive/SuperMarket Analysis.csv")

In [104]:
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             Product line  Unit price  Quantity   Tax 5%     Sales       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

          Time      Payment    cogs  gross margin percentage  gross income  \
0   1:08:00 PM      Ewallet  522.83                 4.761905       26.1415   
1  10:29:00 AM         Cash   76.40                 4.761905        3.8200   
2   1:23:00 PM  Credit card  324.31                 4.761905       16.2155   
3   8:33:00 PM      Ewallet  465.76                 4.761905       23.2880   
4  10:37:00 AM      Ewallet  604.17                 4.761905       30.2085   

   Rating  
0     9.1  
1     9.6  
2     7.4  
3     8.4  
4     5.3

In [105]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Sales                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

* Date and Time are in the wrong Dtype
* Date is in wrong format
* No missing data
* Column Headers need to be clear

In [106]:
df.describe()

Unit price     Quantity       Tax 5%        Sales        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count              1000.000000   1000.000000  1000.00000  
mean                  4.761905     15.379369     6.97270  
std                   0.000000     11.708825     1.71858  
min                   4.761905      0.508500     4.00000  
25%                   4.761905      5.924875     5.50000  
50%                   4.761905     12.088000     7.00000  
75%                   4.761905     22.445250     8.50000  
max                   4.761905     49.650000    10.00000

In [107]:
df.describe(include="O", exclude=["int", "float"])

Invoice ID Branch    City Customer type  Gender         Product line  \
count          1000   1000    1000          1000    1000                 1000   
unique         1000      3       3             2       2                    6   
top     750-67-8428   Alex  Yangon        Member  Female  Fashion accessories   
freq              1    340     340           565     571                  178   

            Date        Time  Payment  
count       1000        1000     1000  
unique        89         506        3  
top     2/7/2019  7:48:00 PM  Ewallet  
freq          20           7      345

* Gross Margin Percentage is 4.761905
* Invoive ID

Both have high cardinality and uniqueness, will be dropped

In [108]:
# fix column names

cols= [col.lower().replace(" ", "_") for col in df.columns]
df.columns= cols
df.head()

invoice_id branch       city customer_type  gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             product_line  unit_price  quantity   tax_5%     sales       date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

          time      payment    cogs  gross_margin_percentage  gross_income  \
0   1:08:00 PM      Ewallet  522.83                 4.761905       26.1415   
1  10:29:00 AM         Cash   76.40                 4.761905        3.8200   
2   1:23:00 PM  Credit card  324.31                 4.761905       16.2155   
3   8:33:00 PM      Ewallet  465.76                 4.761905       23.2880   
4  10:37:00 AM      Ewallet  604.17                 4.761905       30.2085   

   rating  
0     9.1  
1     9.6  
2     7.4  
3     8.4  
4     5.3

In [109]:
# Reformat the date values from / to - then 

df["date"]= pd.to_datetime(df['date'], format = '%m/%d/%Y')
df.date.info()

<class 'pandas.core.series.Series'>
RangeIndex: 1000 entries, 0 to 999
Series name: date
Non-Null Count  Dtype         
--------------  -----         
1000 non-null   datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 7.9 KB


In [110]:
# Reformat the date values from / to - then 

df["time"]= pd.to_datetime(df['time'], format ='%I:%M:%S %p').dt.time
df.time.info()

<class 'pandas.core.series.Series'>
RangeIndex: 1000 entries, 0 to 999
Series name: time
Non-Null Count  Dtype 
--------------  ----- 
1000 non-null   object
dtypes: object(1)
memory usage: 7.9+ KB


In [111]:
df.head()

invoice_id branch       city customer_type  gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             product_line  unit_price  quantity   tax_5%     sales       date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

       time      payment    cogs  gross_margin_percentage  gross_income  \
0  13:08:00      Ewallet  522.83                 4.761905       26.1415   
1  10:29:00         Cash   76.40                 4.761905        3.8200   
2  13:23:00  Credit card  324.31                 4.761905       16.2155   
3  20:33:00      Ewallet  465.76                 4.761905       23.2880   
4  10:37:00      Ewallet  604.17                 4.761905       30.2085   

   rating  
0     9.1  
1     9.6  
2     7.4  
3     8.4  
4     5.3